### PR TITLE
Omniture campaign

### PR DESCRIPTION
**Pull request by umbobabo**
This implementation has the primary aim to enable tracking of campaigns in the WIF 2016.
At the same time the wider aim is the portability of the existing code used for the economist.com into the new plugin+tests approach.

Regarding the .com product this is not the final implementation because there are still part of the original file (assets/ec_omniture_s_code.3.5.js) that need to be ported.

### CODE REVIEW

File: OmnitureThirdPartyPlugins.es6 [hunk 1]
```diff
@@ -0,0 +1,228 @@
+/** Plugin: getQueryParam 2.3 **/
+let s = {};
+/************************** PLUGINS SECTION *************************/
+/* You may insert any plugins you wish to use here.                 */
+
+/** Utility Function: split v1.5 (JS 1.0 compatible)  */
+s.split = new Function("l", "d", ""
+  + "var i,x=0,a=new Array;while(l){i=l.indexOf(d);i=i>-1?i:l.length;a[x"
+  + "++]=l.substring(0,i);l=l.substring(i+d.length);}return a");
```
Comment (keithamus) [inline OmnitureThirdPartyPlugins.es6]: These functions are awful btw. This one, for example, is a reimplementation of `String.prototype.split` except it has loads of really shitty bugs, for example calling it with an empty string as the second argument causes an infinite loop.

I swear I can't even.

A much better implementation of the above code would be `s.split = String.prototype.split.call.bind(String.prototype.split)`. It's a lot less likely to crash on us, and it's smaller, standardised, readable, etc, etc, etc.

Comment (umbobabo) [inline OmnitureThirdPartyPlugins.es6]: Unfortunately they confirmed me that review third party code is out of scope currently.
If you think these can be a serious risk for the stability of the product pls talk with @angelorepole so we can define an acceptable strategy.
Can you pls review all the code.
Tests are still lacking for the OmniturePlugin.

File: OmnitureUtils.es6 [hunk 2]
```diff
@@ -7,34 +78,35 @@ const OmnitureUtils = {
...
-  estFormatDate() {
+  estFormatDate(date) {
     // EST
     const offset = -5.0
-    const clientDate = new Date();
+    const clientDate = (date) ? date : new Date();
     const utc = clientDate.getTime() + (clientDate.getTimezoneOffset() * 60000);
     const serverDate = new Date(utc + (3600000*offset));
     return serverDate;
   },
```
Comment (keithamus) [inline OmnitureUtils.es6]: This is a bad idea. The offsets are coincidental. You'll end up falling all over timezones with this code.

File: OmnitureUtils.es6 [hunk 3]
```diff
@@ -7,34 +78,35 @@ const OmnitureUtils = {
...
     // Returns the time of the event in EST time to the closest half hour.
     // Expected output examples
+    // "7:30AM"
     // "11:00AM"
     // "13:30PM"
-    const date = this.estFormatDate();
+    date = (date) ? this.estFormatDate(date) : this.estFormatDate();
     const hours = date.getHours();
     // Round minutes to 0 or 30.
     let minutes = (Math.floor(date.getMinutes()/30) * 30);
```
Comment (keithamus) [inline OmnitureUtils.es6]: Instead of `estFormatDate` we should send dates in UTC, using `getUTCMinutes()`

**Comment by keithamus** [comment]
To be honest, I don't really know what I am meant to be reviewing. The omniture bits of this code are awful, and are a bomb waiting to go off. I'm surprised half of that code actually works at all.

To be brutally honest, the custom bits of code aren't much better. Everything seems like a hack. Sending custom EST dates, splitting on strings by random delimiters, opaque key names with no context (e.g. `fsrc`, `cid1`, `cid2`). 

If you want me to actually review this code, then I can only say "I can't merge this". There's too much wrong with it, and I should just close the PR.

if you want me to look past all of that, and just sanity check the code - I have no idea what I'm looking at because it's all Omniture incantations, and nothing with any meaning. So I can't do that without spending days trying to understand all of those values.

If this is just a formality, and what you're really asking is for someone to merge it because thats how we do things, then yeah, sure, I can click the merge button and have no accountability of the actual source.

Pick your poison :smile:

**Pull request closed without merging**